integration-tests/src/test/gate-token.test.ts

```diff
@@ -0,0 +1,108 @@
+import { checkAddress, testData, tryRegisterAddress } from 'creditcoin-js/lib/testUtils';
+
+import { extractFee, testIf } from '../utils';
+
+import { deployGATEToken } from '../ctc-deploy';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { Signer, Wallet, Contract } from 'ethers';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { personalSignAccountId, signAccountId } from 'creditcoin-js/lib/utils';
+import { ethSignSignature, personalSignSignature } from 'creditcoin-js/lib/extrinsics/register-address-v2';
+import { requestCollectCoins } from 'creditcoin-js/lib/extrinsics/request-collect-coins';
+import { GATEContract } from 'creditcoin-js/lib/extrinsics/request-collect-coins-v2';
+
+describe('Test GATE Token', (): void => {
+    let ccApi: CreditcoinApi;
+    let sudoSigner: KeyringPair;
+
+    // Needed to interact with the ethererum private node
+    const testingData = testData(
+        (global as any).CREDITCOIN_ETHEREUM_CHAIN as Blockchain,
+        (global as any).CREDITCOIN_CREATE_WALLET,
+    );
+    const { keyring } = testingData;
+    const provider = new JsonRpcProvider((global as any).CREDITCOIN_ETHEREUM_NODE_URL);
+    const deployer = new Wallet((global as any).CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY, provider);
+    const burnAmount = 200;
+
+    // Holds the reference to the deployed GATE contract
+    let gateToken: any;
+    const gateKeyring = new Keyring({ type: 'ed25519', ss58Format: 3 });
+    const gateFaucet = gateKeyring.addFromUri(mnemonicGenerate(12));
+
+    beforeAll(async () => {
+        gateToken = await deployGATEToken(deployer);
+
+        ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+        if ((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY) {
+            sudoSigner = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'lender');
+        }
+    });
+
+    afterAll(async () => {
+        await ccApi.api.disconnect();
+    });
+
+    test('End to end', async () => {
+        const {
+            api,
+            extrinsics: { requestCollectCoinsV2 },
+        } = ccApi;
+
+        await api.tx.sudo
+            .sudo(api.tx.balances.setBalance(gateFaucet.address, 1000, 0))
+            .signAndSend(sudoSigner, { nonce: -1 });
+
+        // Set the on chain location for the burn contract to be the address of the deployer wallet
+        const contract = api.createType('PalletCreditcoinOcwTasksCollectCoinsDeployedContract', {
+            address: gateToken.address,
+            chain: testingData.blockchain,
+        });
+        await api.tx.sudo.sudo(api.tx.creditcoin.setBurnGateContract(contract)).signAndSend(sudoSigner, { nonce: -1 });
+
+        const mintTx = await gateToken.mint(deployer.address, 2500);
+        await mintTx.wait(3);
+        const balance = await gateToken.balanceOf(deployer.address);
+        expect(balance.eq(2500)).toBe(true);
+
+        const burnTx = await gateToken.burn(burnAmount);
+        await burnTx.wait(3);
```

Comment:
Q: what does `.wait(3)` mean?  Why exactly `3` ?

Reply:
That waits for the desired number of confirmations. 3 was arbitrary. On the rust side the transaction will be rejected if there are less than 12 confirmations.

Technically I haven't tested it without the wait so it may be totally unnecessary

Reply:
When burning CTC we call `.wait()` without a numeric argument. I'm not sure what's the default, probably 12. 

I'm not sure if we need to be doing the same here just for consistency or if we can benefit from less confirmations/faster execution time. 

Let's leave this question open for now.

Reply:
Default is 1 IIRC